### PR TITLE
Miniconda base

### PR DESCRIPTION
**Pull request by rgbkrk**
This begins a transition over to using miniconda for Python bits and making our image smaller.

I'd also like to make the only default kernel Python 3. Any objections?

I need to bring libraries back in after I nuked things, so please don't merge.

### CODE REVIEW

**Comment by rgbkrk** [comment]
On a separate note, I'm no longer comfortable with the R package installations because I haven't found a reliable CRAN mirror that uses HTTPS properly.

**Comment by Carreau** [comment]
I can ask Karthik locally on how he does it if you like.

**Comment by Carreau** [comment]
That remind me that figshare is still only over http, and I would like to put some pressure, on them.

**Comment by minrk** [comment]
:+1: to making Python 3 the only default.

**Comment by rgbkrk** [comment]
Apparently `yt` isn't available on conda for Python 3.4

```
Error: Unsatisfiable package specifications.
Generating hint:
[      COMPLETE      ]|##################################################| 100%                            |   0%


Hint: the following combinations of packages create a conflict with the
remaining packages:
  - python 3.4*
  - yt
```

I'm just going to toss `yt` for the moment.

**Comment by rgbkrk** [comment]
This brings us down to `2.875 GB` for the demo image, down from `4.592 GB`.

```
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
jupyter/demo        latest              78ec390bc7cd        10 minutes ago      2.875 GB
jupyter/minimal     latest              a686e82f82a9        18 minutes ago      1.182 GB
```

**Comment by rgbkrk** [comment]
This also cuts the build time a LOT. :smile:

File: Dockerfile [hunk 1]
```diff
@@ -1,104 +1,48 @@
...
-RUN echo "library(devtools); install_github('jimhester/robustr'); install_github('jimhester/covr')" | R --no-save
-
-RUN echo "install.packages(c('base64enc', 'Cairo', 'codetools', 'data.table', 'gridExtra', 'gtable', 'hexbin', 'jpeg', 'Lahman', 'lattice'))" | R --no-save
-RUN echo "install.packages(c('MASS', 'PKI', 'png', 'microbenchmark', 'mgcv', 'mapproj', 'maps', 'maptools', 'mgcv', 'multcomp', 'nlme'))" | R --no-save
-RUN echo "install.packages(c('nycflights13', 'quantreg', 'rJava', 'roxygen2', 'RSQLite', 'XML'))" | R --no-save
-
-# Workaround for issue with ADD permissions
-USER root
-ADD common/profile_default /home/jovyan/.ipython/profile_default
-RUN cp /home/jovyan/.ipython/profile_default/static/custom/* /srv/ipython/IPython/html/static/custom/ && chmod a+r /srv/ipython/IPython/html/static/custom/
```
Comment (rgbkrk) [inline Dockerfile]: I missed adding this one back in.

**Pull request merged**